File: SDKs/github.com/ghes-3.0/1.1.4/go/pkg/models/operations/reposlistwebhooks.go
package operations

import (
	"openapi/pkg/models/shared"
)

type ReposListWebhooksPathParams struct {
	Owner string `pathParam:"style=simple,explode=false,name=owner"`
	Repo  string `pathParam:"style=simple,explode=false,name=repo"`
}

type ReposListWebhooksQueryParams struct {
	Page    *int64 `queryParam:"style=form,explode=true,name=page"`
	PerPage *int64 `queryParam:"style=form,explode=true,name=per_page"`
}

type ReposListWebhooksRequest struct {
	PathParams  ReposListWebhooksPathParams
	QueryParams ReposListWebhooksQueryParams
}

type ReposListWebhooksResponse struct {
	ContentType string
	Headers     map[string][]string
	StatusCode  int64
	BasicError  *shared.BasicError
	Hooks       []shared.Hook
}

File: SDKs/github.com/ghes-2.21/1.1.4/typescript/src/sdk/models/operations/reposlistwebhooks.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class ReposListWebhooksPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=owner" })
  owner: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=repo" })
  repo: string;
}


export class ReposListWebhooksQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}


export class ReposListWebhooksRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: ReposListWebhooksPathParams;

  @Metadata()
  queryParams: ReposListWebhooksQueryParams;
}


export class ReposListWebhooksResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  headers: Map<string, string[]>;

  @Metadata()
  statusCode: number;

  @Metadata()
  basicError?: shared.BasicError;

  @Metadata({ elemType: shared.Hook })
  hooks?: shared.Hook[];
}
